subsys/instrumentation/transport/uart.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 Linaro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/reboot.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/instrumentation/instrumentation.h>
+
+#define COMMAND_BUFFER_SIZE 32
+char _cmd_buffer[COMMAND_BUFFER_SIZE];
+
+__no_instrumentation__
+void handle_cmd(char *cmd)
+{
+	char *beginptr;
+	char *endptr;
+	long address;
+
+	if (strcmp("reboot", cmd) == 0) {
+		sys_reboot(SYS_REBOOT_COLD);
+	} else if (strcmp("status", cmd) == 0) {
+		printk("%d %d\n", instr_tracing_supported(), instr_profiling_supported());
+	} else if (strcmp("ping", cmd) == 0) {
+		printk("pong\n");
+	} else if (strcmp("dump_trace", cmd) == 0) {
+		instr_dump_buffer_uart();
+	} else if (strcmp("dump_profile", cmd) == 0) {
+		instr_dump_deltas_uart();
+	} else if (strncmp(cmd, "trigger", strlen("trigger")) == 0) {
+		beginptr = cmd + strlen("trigger");
+		address = strtol(beginptr, &endptr, 16);
+		if (endptr != beginptr) {
+			instr_set_trigger_func((void *)address);
+		} else {
+			printk("invalid argument for 'trigger' in: '%s'\n", cmd);
+		}
+	} else if (strncmp(cmd, "stopper", strlen("stopper")) == 0) {
+		beginptr = cmd + strlen("stopper");
+		address = strtol(beginptr, &endptr, 16);
+		if (endptr != beginptr) {
+			instr_set_stop_func((void *)address);
+		} else {
+			printk("invalid argument for 'stopper' in: '%s'\n", cmd);
```

Comment:
Correct me if I'm wrong on a usage model here.  I envisioning an automated usage here that would have UART scraping as part of the process to generate profiling/tracing trends over time.  If that is indeed the case, does it make sense to keep these kinds of messages all the same formatting to enable scraping tools easier?  For example: `stopper: invalid argument in %s` would match what is going on for line 65